sympy/physics/mechanics/tests/test_models.py

```diff
@@ -0,0 +1,200 @@
+
+    kane3 = models.multi_mass_spring_damper(1, True, True)
+    symexpr3 = Matrix([[v0], [(-c0*v0 + g*m0 - k0*x0 + f0)/m0]])
+    assert simplify(symexpr3 - kane3.rhs()) == Matrix([0, 0])
+
+    kane4 = models.multi_mass_spring_damper(1, False, True)
+    symexpr4 = Matrix([[v0], [(-c0*v0 - k0*x0 + f0)/m0]])
+    assert simplify(symexpr4 - kane4.rhs()) == Matrix([0, 0])
+
+
+def test_multi_mass_spring_damper_higher_order():
+    c0, k0, m0 = symbols("c0 k0 m0")
+    c1, k1, m1 = symbols("c1 k1 m1")
+    c2, k2, m2 = symbols("c2 k2 m2")
+    v0, x0 = dynamicsymbols("v0 x0")
+    v1, x1 = dynamicsymbols("v1 x1")
+    v2, x2 = dynamicsymbols("v2 x2")
+
+    kane1 = models.multi_mass_spring_damper(3)
+    symexpr = Matrix([[v0], [v1], [v2], [(-c0*v0 - k0*x0 - m2*(-c2*v2 - k2*x2 -
+                     m2*(-c0*v0 - k0*x0)/(m0 + m1 + m2) - (-m2*(m1 + m2)/(m0 +
+                     m1 + m2) + m2)*(-c1*v1 - k1*x1 - (m1 + m2)*(-c0*v0 -
+                     k0*x0)/(m0 + m1 + m2))/(m1 + m2 - (m1 + m2)**2/(m0 + m1 +
+                     m2)))/(-m2**2/(m0 + m1 + m2) + m2 - (-m2*(m1 + m2)/(m0 + m1
+                     + m2) + m2)**2/(m1 + m2 - (m1 + m2)**2/(m0 + m1 + m2))) -
+                     (m1 + m2)*(-c1*v1 - k1*x1 - (m1 + m2)*(-c0*v0 - k0*x0)/(m0
+                     + m1 + m2) - (-m2*(m1 + m2)/(m0 + m1 + m2) + m2)*(-c2*v2 -
+                     k2*x2 - m2*(-c0*v0 - k0*x0)/(m0 + m1 + m2) - (-m2*(m1 +
+                     m2)/(m0 + m1 + m2) + m2)*(-c1*v1 - k1*x1 - (m1 + m2)*
+                     (-c0*v0 - k0*x0)/(m0 + m1 + m2))/(m1 + m2 - (m1 + m2)**2/
+                     (m0 + m1 + m2)))/(-m2**2/(m0 + m1 + m2) + m2 - (-m2*(m1 +
+                     m2)/(m0 + m1 + m2) + m2)**2/(m1 + m2 - (m1 + m2)**2/(m0 +
+                     m1 + m2))))/(m1 + m2 - (m1 + m2)**2/(m0 + m1 + m2)))/(m0 +
+                     m1 + m2)], [(-c1*v1 - k1*x1 - (m1 + m2)*(-c0*v0 - k0*x0)/
+                     (m0 + m1 + m2) - (-m2*(m1 + m2)/(m0 + m1 + m2) + m2)*(-c2*
+                     v2 - k2*x2 - m2*(-c0*v0 - k0*x0)/(m0 + m1 + m2) - (-m2*(m1
+                     + m2)/(m0 + m1 + m2) + m2)*(-c1*v1 - k1*x1 - (m1 + m2)*
+                     (-c0*v0 - k0*x0)/(m0 + m1 + m2))/(m1 + m2 - (m1 + m2)**2/
+                     (m0 + m1 + m2)))/(-m2**2/(m0 + m1 + m2) + m2 - (-m2*(m1 +
+                     m2)/(m0 + m1 + m2) + m2)**2/(m1 + m2 - (m1 + m2)**2/(m0 +
+                     m1 + m2))))/(m1 + m2 - (m1 + m2)**2/(m0 + m1 + m2))],
+                     [(-c2*v2 - k2*x2 - m2*(-c0*v0 - k0*x0)/(m0 + m1 + m2) -
+                     (-m2*(m1 + m2)/(m0 + m1 + m2) + m2)*(-c1*v1 - k1*x1 - (m1 +
+                     m2)*(-c0*v0 - k0*x0)/(m0 + m1 + m2))/(m1 + m2 - (m1 +
+                     m2)**2/(m0 + m1 + m2)))/(-m2**2/(m0 + m1 + m2) + m2 - (-m2*
+                     (m1 + m2)/(m0 + m1 + m2) + m2)**2/(m1 + m2 - (m1 + m2)**2/
+                     (m0 + m1 + m2)))]])
+    assert simplify(symexpr - kane1.rhs()) == Matrix([0, 0, 0, 0, 0, 0])
```

Comment:
Maybe this is taking a long time. `trigsimp` might be better to explicitly call.